temporalio/worker/_workflow.py

```diff
@@ -250,9 +263,14 @@ async def _handle_activation(
                         activate_task, self._deadlock_timeout_seconds
                     )
                 except asyncio.TimeoutError:
-                    raise RuntimeError(
-                        f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
+                    msg, tb = self._build_deadlock_exception(
+                        workflow, self._deadlock_timeout_seconds
                     )
+
+                    if tb:
+                        raise DeadlockError(msg, tb) from None
+                    raise DeadlockError(msg) from None
```

Comment:
```suggestion
                    raise self._build_deadlock_error(workflow) from None
```

You can access deadlock timeout seconds inside the function instead of making it static and you can build the error in the function, no need to split the logic. Or maybe even better:

```suggestion
                    raise DeadlockError.from_deadlocked_workflow(workflow, self._deadlock_timeout_seconds) from None
```

That way all this extra code isn't on this worker class but on the deadlock error class. But it doesn't really matter.